### ISSUE TITLE
BHoM_UI: DataAccessor implementations create a closed inheritance loop

### ISSUE THREAD

**Issue by epignatelli** [BHoM/BHoM_UI#71]
In the Grasshopper implementation of the `DataAccessor`, the `DataAccessor_GH` owns a `GH_Component`, and the `CallerComponent` owns a `DataAccessor_GH`. 

This closed loop makes serialisation of a `CallerComponent` to fail.

It may required modifications in the implemented UIs as well, so it's may be not circumscribed to the BHoM_UI

**Comment by adecler** [comment]
@epignatelli , shouldn't that be an issue in Grasshopper_Toolkit ?

**Comment by epignatelli** [comment]
I don't think it is a Grasshopper issue. A `DataAccessor` in Grasshopper already holds a `GH_Component`, I am not sure there is something we can do about it.

In the `Caller` we add to it the inverse, a `Caller` holding a `DataAccessor`. That's the only place I can see we have space to action.

**Comment by adecler** [comment]
@epignatelli , I have fixed this in Grasshopper so moving the issue to that repo.
